rnd/autogpt_server/autogpt_server/migrations/20240604091708_init/migration.sql

```diff
@@ -0,0 +1,101 @@
+-- CreateTable
+CREATE TABLE "AgentGraph" (
+    "id" TEXT NOT NULL PRIMARY KEY,
+    "name" TEXT,
+    "description" TEXT,
+    "startingAgentNodeId" TEXT NOT NULL,
+    CONSTRAINT "AgentGraph_startingAgentNodeId_fkey" FOREIGN KEY ("startingAgentNodeId") REFERENCES "AgentNode" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+-- CreateTable
+CREATE TABLE "AgentNode" (
+    "id" TEXT NOT NULL PRIMARY KEY,
+    "agentComponentId" TEXT NOT NULL,
+    "agentGraphId" TEXT NOT NULL,
+    CONSTRAINT "AgentNode_agentComponentId_fkey" FOREIGN KEY ("agentComponentId") REFERENCES "AgentComponent" ("id") ON DELETE RESTRICT ON UPDATE CASCADE,
+    CONSTRAINT "AgentNode_agentGraphId_fkey" FOREIGN KEY ("agentGraphId") REFERENCES "AgentGraph" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+-- CreateTable
+CREATE TABLE "AgentNodeTrigger" (
+    "id" TEXT NOT NULL PRIMARY KEY,
+    "agentNodeId" TEXT NOT NULL,
+    "triggerNodeId" TEXT NOT NULL,
+    "triggerOutputId" TEXT NOT NULL,
+    CONSTRAINT "AgentNodeTrigger_agentNodeId_fkey" FOREIGN KEY ("agentNodeId") REFERENCES "AgentNode" ("id") ON DELETE RESTRICT ON UPDATE CASCADE,
+    CONSTRAINT "AgentNodeTrigger_triggerNodeId_fkey" FOREIGN KEY ("triggerNodeId") REFERENCES "AgentNode" ("id") ON DELETE RESTRICT ON UPDATE CASCADE,
+    CONSTRAINT "AgentNodeTrigger_triggerOutputId_fkey" FOREIGN KEY ("triggerOutputId") REFERENCES "AgentComponentOutput" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+-- CreateTable
+CREATE TABLE "AgentComponent" (
+    "id" TEXT NOT NULL PRIMARY KEY,
+    "name" TEXT NOT NULL,
+    "code" TEXT NOT NULL,
+    "inputSchema" TEXT NOT NULL
+);
```

Comment:
This is exactly the same name as in `autogpt` and may be confusing to users as long as we use `autogpt` with components in Agent Server. Even if we stick to `Component` how about `GraphComponent`? The same applies to all others `Agent*`, e.g. `GraphComponentOutput`. Maybe except `AgentGraph`, this could be `NodeGraph`.

Reply:
@kcze These files are autogenerated. It's better to add your review comments to `schema.prisma`

Reply:
Renamed to AgentBlock